Formula/zsh-syntax-highlighting.rb

```diff
@@ -13,6 +13,14 @@ class ZshSyntaxHighlighting < Formula
   end
 
   def install
+    inreplace "Makefile" do |s|
+      # Fix for `brew install --git zsh-syntax-highlighting`.
+      # Without this, the git repository with no HEAD would cause
+      # `git rev-parse` to fail and abort the installation.
+      s.gsub! "git rev-parse HEAD",
```

Comment:
Can you please submit this patch to the upstream developers of this project and add a link to the upstream patch submission and explanation of why the patch is needed in a comment in the formula file. Thanks!